### PR TITLE
Add all filter messages from server

### PR DESCRIPTION
**Pull request by petterarvidsson**
- Add a Filter abstract base class
- Add both messages and their corresponding filter containers

### CODE REVIEW

**Comment by petterarvidsson** [comment]
this is WIP, documentation is still to be done.

**Comment by petterarvidsson** [comment]
@nsg Please have a look at the filter stuff extracted from the server and its documentation.

File: src/main/java/konstructs/api/messages/InteractTertiaryFilter.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        return new InteractTertiaryFilter(chain, getMessage());
+    }
+
+    @Override
+    protected Filter<InteractTertiary> nextFilter(ActorRef[] chain, InteractTertiary message) {
+        return new InteractTertiaryFilter(chain, message);
+    }
+
+    @Override
+    public void drop(ActorRef sender) {
```
Comment (nsg) [inline src/main/java/konstructs/api/messages/InteractTertiaryFilter.java:45]: (not specific for this function, but a general question about all `@Override` with no comments, like this one.)

Maybe add a short explanation and/or link the other `drop`? Easier for the user browsing the documentation, what do you think?

**Comment by nsg** [comment]
With a minor note, I have nothing more than say than ... nice! Good and easy to understand documentation! I like it :+1:

File: src/main/java/konstructs/api/messages/InteractTertiaryFilter.java (hunk 1 above)
Comment (petterarvidsson) [inline src/main/java/konstructs/api/messages/InteractTertiaryFilter.java:45]: @nsg The docs from the base class is already included by the javadoc tool automatically.

**Pull request merged**